compiler/rustc_data_structures/src/sync.rs

```diff
@@ -143,6 +143,11 @@ cfg_if! {
                 self.0.set(val);
                 result
             }
+            pub fn fetch_and(&self, val: bool, _: Ordering) -> bool {
+                let result = self.0.get() & val;
+                self.0.set(val);
+                result
```

Comment:
This currently does `self.swap(val) & val`, not `self.fetch_and(val)`. Same for the "`fetch_or`" above.

```suggestion
                let old = self.0.get();
                self.0.set(old & val);
                old
```